commcare_connect/opportunity/models.py

```
@@ -676,6 +686,39 @@ def __setattr__(self, name, value):
                 self.status_modified_date = now()
         super().__setattr__(name, value)
 
+    def save(self, rank=1, **kwargs):
+        # The code below automatically switches/create CompletedWork when the UNIQUE_USER_VISIT_CONSTRAINT
+        # is triggered.
+        try:
+            with transaction.atomic():
+                super().save(**kwargs)
+        except IntegrityError as e:
+            if UNIQUE_USER_VISIT_CONSTRAINT not in str(e):
+                raise e
```

Ruff (0.14.3)

Use `raise` without specifying exception name

Remove exception name

(TRY201)